src/test/java/htsjdk/samtools/CRAMComplianceTest.java

```diff
@@ -265,6 +295,80 @@ public void testBAMThroughCRAMRoundTrip() throws IOException, NoSuchAlgorithmExc
         }
     }
 
+//    @Test
+//    public void testCompressionWithAllAmbiguityCodes() {
+//        createAmbiguityCodeTestSAM();
+//    }
+
+    @Test
+    public void testRoundTripSAMWithAllAmbiguityCodesThroughCRAM() throws IOException {
+        final File inputSamFile = new File("src/test/resources/htsjdk/samtools/cram/", "allReferenceBases.sam");
+        final File inputReference = new File("src/test/resources/htsjdk/samtools/cram/", "allReferenceBases.fasta");
+        final SAMFileHeader samHeader = getFileHeader(inputSamFile, inputReference);
+
+        List<SAMRecord> originalSAMRecords = getSAMRecordsFromFile(inputSamFile, inputReference);
+        final File tempCRAMFile = File.createTempFile("roundTripAmbiguityCodes", CramIO.CRAM_FILE_EXTENSION);
+        tempCRAMFile.deleteOnExit();
+
+        writeRecordsToFile(
+                originalSAMRecords,
+                tempCRAMFile,
+                inputReference,
+                samHeader
+        );
+
+        List<SAMRecord> roundTripRecords = getSAMRecordsFromFile(tempCRAMFile, inputReference);
+        for (int i = 0; i < originalSAMRecords.size(); i++) {
+            final SAMRecord originalSAMRec = originalSAMRecords.get(i);
+            final SAMRecord roundTripCRAMRec = roundTripRecords.get(i);
+
+            //if (!originalSAMRec.equals(roundTripRecords.get(i))) {
+            if (!Arrays.equals(originalSAMRec.getReadBases(), roundTripCRAMRec.getReadBases())) {
```

Comment:
similar logging block, are these important to the tests somehow or just for debugging?  Another ignored equals result as well.